RadClass/RadClass.py

```diff
@@ -0,0 +1,205 @@
+        self.processor.init_database(self.filename, self.datapath)
+        # parameters for keeping track of progress through a file
+
+        self.working_time = self.processor.timestamps[0]
+
+    def collapse_data(self, rows):
+        '''
+        Integrates a subset of data from the total data matrix given some
+        integration time. Utilizes data_slice() from DataSet to extract
+        the requisite rows of data and then numpy.sum(data,axis=0) to combine
+        all rows. Assumes negligible drift and accurate energy calibration.
+
+        NOTE: Returned data is normalized for live times (1s measurement
+        period less dead time). i.e. The returned data is a count rate at
+        each bin.
+
+        Return: 1D numpy array for integration interval
+        '''
+
+        # extract requisite data rows
+        data_matrix = self.processor.data_slice(self.datapath, rows)
+
+        # normalize by live times to produce count rate data
+        # processor.live can be indexed by appropriate timestamps but
+        # data_matrix only has indices for rows.
+        data_matrix = data_matrix / self.processor.live[rows][:, None]
+
+        # utilizes numpy architecture to sum data
+        total = np.sum(data_matrix, axis=0)
+
+        return total
+
+    def collect_rows(self):
+        '''
+        Uses the integration duration to collect all rows needed for analysis.
+        Uses numpy.arange() to enumerate row indices from start to end.
+
+        Return: list of rows for data indexing
+        '''
+
+        # collect start index from tracked timestamp
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+
+        # NOTE: this behavior is currently disabled by the if-statement in
+        # march() removing the 'or' portion with allow this to work
+        #
+        # if the final portion of the file is smaller than a full integration
+        # interval, only what is left is collected for this analysis
+        end_i = min(start_i + self.integration,
+                    len(self.processor.timestamps) - 1)
+
+        # enumerate number of rows to integrate exclusive of the endpoint
+        rows = np.arange(start_i, end_i)
+        return rows
+
+    def march(self):
+        '''
+        Advance a step using stride time to find the new working integration
+        interval. Also keeps track of the EOF state but checking for
+        out-of-index situations.
+
+        Returns: boolean for EOF
+        '''
+
+        # find the working interval starting index and advance stride
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+        new_i = start_i[0] + self.stride
+
+        # stop analysis if EOF reached
+        # NOTE: stops prematurely, for windows of full integration only
+        running = True
+        if new_i >= (len(self.processor.timestamps) or
+                     (new_i + self.integration) >=
+                     len(self.processor.timestamps)):
```

Comment:
I don't think the parentheses are right here (maybe this was my fault for the code suggestion?)